### DISCUSSION TITLE
Building in Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by AdMorgado** [FS-Online/Driverless-Competition-Simulator#295] [General]
Hey, I was able to build and run the simulator in Ubuntu 22.04 for ROS 2 Humble, but I had to make a few tweaks myself (mostly just changing compiler version and adding ros2 package).
Are there any official plans to allow building and running out of the box?

Thanks.

**Comment by wouter-heerwegh** [comment]
Hi @AdMorgado,

It's a bit of a hassle to maintain versions of the simulator for all ROS distributions. I think it's best to focus on distributions that have not reached the end of life. For instance, for ROS1 melodic, it runs out of the box. A small change is needed to run it in noetic.

With regards to ROS2, the current version is compilable out of the box with galactic, could also be with foxy (haven't checked), but galactic will reach the end of life in November. We haven't switched to humble because it's still fairly new (officially released 23rd of May 2022), which means that there is a big chance not a lot of other ros packages do not support humble yet. So it makes more sense to still stay on foxy/galactic.

If you have a good idea of how we could properly support out of the box running for multiple ros versions, feel free to suggest them.

In the meantime you can already open a pull request with your changes, so that when we switch to humble, it should just be a merge of a pull request.

**Comment by b1n-ch1kn** [comment]
@AdMorgado What were the tweaks you had to make on your Ubuntu 22.04 install? My team is considering the switch to Humble (in fact, we already have, just on 20.04 so our hardware drivers were compatible), but if you were able to change a few things to get 22.04 compatibility for the sim, it could make our eventual switch to 22.04 dev environments easier. 
Thanks!

**Reply by AdMorgado** [reply]
You can check the small commits I've executed on my fork: https://github.com/AdMorgado/Formula-Student-Driverless-Simulator/
Although I wouldn't recommend you use my fork, it's already out of date and the fixes were just on the ROS side to get the Airsim dependency to build and fix some "galactic to humble" errors. Regardless it worked well for me and my team. 
Cheers.

**Reply by b1n-ch1kn** [reply]
Awesome. Actually you found the same issues I had for galactic-to-humble here: https://github.com/FS-Driverless/Formula-Student-Driverless-Simulator/pull/317.  I just removed the deps where they were included as they weren't in use. Good to keep in mind for the future if they are re-added and used.
Will have a look at the newer clang and see if we can still have backwards-compatibility without ditching old clang.
Glad to hear it worked, hopefully we can get a compatible out-of-box solution, otherwise a Humble-only branch could be an option. 
Thanks

**Comment by b1n-ch1kn** [comment]
Hi @AdMorgado, I did some testing using clang 12 and C++17, which are supported in Ubuntu 22.04 and 20.04 so Galactic and Humble were able to run the sim on their respective OS's. 

Found some issues with the ROS2 packages `CMakeLists.txt` and their pinned minimum CMake version. The FS msgs and bridge required CMake 3.5 minimum and despite having higher CMake installed locally (3.21), it was having issues until I pinned the minimum versions to 3.18 - still works in 20.04 and 22.04.
Feel free to check out my PR https://github.com/FS-Driverless/Formula-Student-Driverless-Simulator/pull/327.
